base58/base58.go

```diff
@@ -21,8 +19,8 @@ func Decode(b string) []byte {
 	j := big.NewInt(1)
 
 	for i := len(b) - 1; i >= 0; i-- {
-		tmp := strings.IndexAny(alphabet, string(b[i]))
-		if tmp == -1 {
+		tmp := b58[b[i]]
+		if tmp == 255 {
 			return []byte("")
 		}
 		idx := big.NewInt(int64(tmp))
```

Comment:
I know this was already there, but while you're in the mode of optimizing, there is no need to create a new big integer each time here.  

```Go

var scratch *big.Int
for i := len(b) - 1; i >= 0; i-- {
	...
	scratch.Mul(j, idx)
	answer.Add(answer, scratch)
	j.Mul(j, bigRadix)
}
```